docs/appendix-A.rst

```diff
@@ -0,0 +1,1783 @@
+
+.. literalinclude:: ../regression/run/clibrary/clibrary.c
+   :language: c
+   :start-after: start PassByValue
+   :end-before: end PassByValue
+
+YAML:
+
+.. code-block:: yaml
+
+    - decl: double PassByValue(double arg1, int arg2)
+
+Both types are supported directly by the ``iso_c_binding`` module
+so there is no need for a Fortran function.
+The C function can be called directly by the Fortran interface
+using the ``bind(C)`` keyword.
+
+Calls C via the interface:
+
+.. literalinclude:: ../regression/reference/clibrary/wrapfclibrary.f
+   :language: fortran
+   :start-after: start pass_by_value
+   :end-before: end pass_by_value
+   :dedent: 8
+
+Fortran usage:
+
+.. code-block:: fortran
+
+    real(C_DOUBLE) :: rv_double
+    rv_double = pass_by_value(1.d0, 4)
+    call assert_true(rv_double == 5.d0)
+
+.. ############################################################
+
+.. _example_PassByReference:
+
+PassByReference
+^^^^^^^^^^^^^^^
+
+C library function:
+
+.. literalinclude:: ../regression/run/clibrary/clibrary.c
+   :language: c
+   :start-after: start PassByReference
+   :end-before: end PassByReference
+
+YAML:
+
+.. code-block:: yaml
+
+    - decl: void PassByReference(double *arg1+intent(in), int *arg2+intent(out))
+
+Calls C via the interface:
+
+.. literalinclude:: ../regression/reference/clibrary/wrapfclibrary.f
+   :language: fortran
+   :start-after: start pass_by_reference
+   :end-before: end pass_by_reference
+   :dedent: 8
+
+Example usage:
+
+.. code-block:: fortran
+
+    integer(C_INT) var
+    call pass_by_reference(3.14d0, var)
+    call assert_equals(3, var)
+
+.. ############################################################
+
+.. _example_Sum:
+
+Sum
+^^^
+
+C++ library function:
+
+.. literalinclude:: ../regression/run/pointers/pointers.cpp
+   :language: c
+   :start-after: start Sum
+   :end-before: end Sum
+
+:file:`pointers.yaml`:
+
+.. code-block:: yaml
+
+   - decl: void Sum(int len +implied(size(values)),
+                    int *values +dimension(:)+intent(in),
+                    int *result +intent(out))
+
+:file:`wrappointers.cpp`:
+
+.. literalinclude:: ../regression/reference/pointers/wrappointers.cpp
+   :language: c
+   :start-after: start POI_sum
+   :end-before: end POI_sum
+
+Called via the Fortran interface:
+
+.. literalinclude:: ../regression/reference/pointers/wrapfpointers.f
+   :language: fortran
+   :start-after: start c_sum
+   :end-before: end c_sum
+   :dedent: 8
+
+The Fortran wrapper:
+
+.. literalinclude:: ../regression/reference/pointers/wrapfpointers.f
+   :language: fortran
+   :start-after: start sum
+   :end-before: end sum
+   :dedent: 4
+
+Example usage:
+
+.. code-block:: fortran
+
+    integer(C_INT) rv_int
+    call sum([1,2,3,4,5], rv_int)
+    call assert_true(rv_int .eq. 15, "sum")
+
+
+.. ############################################################
+
+.. _example_getMinMax:
+
+getMinMax
+^^^^^^^^^
+
+No Fortran function is created.  Only an interface to a C wrapper
```

Comment:
Please clarify -- I don't understand this sentence... there is a Fortran function below.

I think you are saying that there is no Fortran function with references?

Reply:
The code was in a Fortran interface block (a.k.a prototype). So it does not have executable code. I fixed up the literalinclude2 option to add explicit interface/end interface statements around the interfaces.  This makes them appear in the document and give better context of what the literalinclude section is about.